samples/tiled-convolution/tiled-convolution.cpp

```diff
@@ -0,0 +1,307 @@
+/***************************************************************************
+ *
+ *  Copyright (C) 2018 Codeplay Software Limited
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  For your convenience, a copy of the License has been included in this
+ *  repository.
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  Codeplay's ComputeCpp SDK
+ *
+ *  tiled-convolution.cpp
+ *
+ *  Description:
+ *    Sample code that illustrates how to divide data into tiles and lauch
+ *    separate kerenl per tile by using range accessors and parallel for with
+ *    offset in SYCL.
+ *
+ **************************************************************************/
+#include <SYCL/sycl.hpp>
+#include <chrono>
+#include <ctime>
+#include <iostream>
+
+// configuring the openCL local size
+template <typename index_t>
+struct opencl_configuration_t {
+  static constexpr index_t local_size_m = 1;
+  static constexpr index_t local_size_n = 32;
+};
+
+// define the 2-d matrix size
+template <typename index_t>
+class matrix_size_t {
+ public:
+  const index_t m;
+  const index_t n;
+  matrix_size_t(const index_t m_, const index_t n_) : m(m_), n(n_) {}
+  inline index_t size() const { return (m * n); }
+};
+
+// the tiled based convolution functor
+template <typename read_accessor_t, typename write_accessor_t, typename index_t,
+          typename mat_size_t>
+class conv {
+ private:
+  read_accessor_t fil_acc;
+  read_accessor_t in_acc;
+  write_accessor_t out_acc;
+  const mat_size_t total_size;
+  const mat_size_t fil_size;
+
+ public:
+  // constructing the functor
+  conv(read_accessor_t fil_acc_, read_accessor_t in_acc_,
+       write_accessor_t out_acc_, const mat_size_t total_size_,
+       const mat_size_t fil_size_)
+      : fil_acc(fil_acc_),
+        in_acc(in_acc_),
+        out_acc(out_acc_),
+        total_size(total_size_),
+        fil_size(fil_size_) {}
+  void inline operator()(cl::sycl::nd_item<2> item_id) {
+    index_t id_m = item_id.get_global(0);
+    index_t id_n = item_id.get_global(1);
+    index_t m, f_m;
+    index_t n, f_n;
+    typename write_accessor_t::value_type val = 0.0;
+#ifdef __SYCL_DEVICE_ONLY__
+#pragma umroll
```

Comment:
are you sure is "umroll" ? :-) i think its unroll. 
For which platform do you need this unroll? or is it beneficial across all range of CE platforms?

Reply:
fixed